### DISCUSSION TITLE
Test.cls custom Attribute

### DISCUSSION THREAD

**Discussion by Beakerboy** [sancarn/stdVBA#90] [General]
Line 11 of tests/lib/Test.cls says

`Attribute shTests.VB_VarHelpID = -1`

I was reading through the VBA 7.1 specification and it does not say anything about allowing custom attributes. I was wondering if you could tell me the purpose of this, and Is there any documentation that this is allowed? I’ve been filing a ton of big reports against with Microsoft about things missing or incorrect In the document and I was wondering if this is another.

I was using your files to test out my Linting tool as well as a precompiler and came across this as something that broke the parser.

https://msopenspecs.azureedge.net/files/MS-VBAL/%5bMS-VBAL%5d.pdf

**Comment by sancarn** [comment]
Interesting! It's something which is exported by Excel when exporting a sheet to a cls file. Unaltered otherwise. Not sure if it has any meaning or not unfortunately. Thanks for reporting the issues to Microsoft, and thanks for making that linter! It will be really useful going forward! :)

**Reply by Beakerboy** [reply]
I found some non-Microsoft sites that explained what they are for and filed a bug report with Microsoft against their documentation.
